lib/mbr.ml

```diff
@@ -71,6 +71,8 @@ module Partition = struct
   let size_sectors t = Int64.(logand (of_int32 t.sectors) 0xFFFF_FFFFL)
 
   let make ?(active = false) ?(ty = 6) first_absolute_sector_lba sectors =
+    (* ty has to fit in a uint8_t, and ty=0 is reserved for empty partition entries *)
+    assert (ty > 0 && ty < 256);
```

Comment:
Would prefer an Invalid_argument exception be raised here, rather than an assertion failure. (or an `Error`)